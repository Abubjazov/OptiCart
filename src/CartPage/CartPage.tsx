import { CartList, Checkout, Header } from '../components'
import './CartPage.scss'

export const CartPage = () => (
	<div className='cart'>
		<Header />
		<CartList />
		<Checkout />
	</div>
)
