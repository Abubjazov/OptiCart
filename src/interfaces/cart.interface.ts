import { Product } from './product.interface'

export enum CartActionTypes {
	FETCH_CART = 'FETCH_CART',
	FETCH_CART_SUCCESS = 'FETCH_CART_SUCCESS',
	FETCH_CART_ERROR = 'FETCH_CART_ERROR',
	ADD_TO_CART = 'ADD_TO_CART',
	ADD_TO_CART_SUCCESS = 'ADD_TO_CART_SUCCESS',
	ADD_TO_CART_ERROR = 'ADD_TO_CART_ERROR',
	REMOVE_FROM_CART = 'REMOVE_FROM_CART',
	REMOVE_FROM_CART_SUCCESS = 'REMOVE_FROM_CART_SUCCESS',
	REMOVE_FROM_CART_ERROR = 'REMOVE_FROM_CART_ERROR',
	UPDATE_CART_QUANTITY = 'UPDATE_CART_QUANTITY',
	UPDATE_CART_QUANTITY_SUCCESS = 'UPDATE_CART_QUANTITY_SUCCESS',
	UPDATE_CART_QUANTITY_ERROR = 'UPDATE_CART_QUANTITY_ERROR',
}

export interface CartListItem extends Product {
	product_id: number
	quantity: number
}

export interface CartState {
	cart: CartListItem[]
	currentItemId: null | number
	status: 'waiting' | 'error' | 'loading'
	error: null | string
}

interface FetchCartAction {
	type: CartActionTypes.FETCH_CART
}

interface FetchCartSuccessAction {
	type: CartActionTypes.FETCH_CART_SUCCESS
	payload: CartListItem[]
}

interface FetchCartErrorAction {
	type: CartActionTypes.FETCH_CART_ERROR
	payload: string
}

interface addToCartAction {
	type: CartActionTypes.ADD_TO_CART
	payload: number
}

interface addToCartSuccessAction {
	type: CartActionTypes.ADD_TO_CART_SUCCESS
	payload: CartListItem[]
}
interface addToCartErrorAction {
	type: CartActionTypes.ADD_TO_CART_ERROR
	payload: string
}
// interface Action {
// 	type: CartActionTypes
// 	payload: any
// }
// interface Action {
// 	type: CartActionTypes
// 	payload: any
// }
// interface Action {
// 	type: CartActionTypes
// 	payload: any
// }
// interface Action {
// 	type: CartActionTypes
// 	payload: any
// }
// interface Action {
// 	type: CartActionTypes
// 	payload: any
// }
// interface Action {
// 	type: CartActionTypes
// 	payload: any
// }

export type CartAction =
	| FetchCartAction
	| FetchCartSuccessAction
	| FetchCartErrorAction
	| addToCartAction
	| addToCartSuccessAction
	| addToCartErrorAction
