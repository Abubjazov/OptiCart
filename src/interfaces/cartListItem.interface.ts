import { Product } from './product.interface'

export enum CartActionTypes {
	FETCH_CART = 'FETCH_CART',
	FETCH_CART_SUCCESS = 'FETCH_CART_SUCCESS',
	FETCH_CART_ERROR = 'FETCH_CART_ERROR',
}

export interface CartListItem extends Product {
	product_id: number
	quantity: number
}

export interface CartState {
	cart: CartListItem[]
	status: 'waiting' | 'error' | 'loading'
	error: null | string
}

interface FetchCartAction {
	type: CartActionTypes.FETCH_CART
}

interface FetchCartSuccessAction {
	type: CartActionTypes.FETCH_CART_SUCCESS
	payload: CartListItem[]
}

interface FetchCartErrorAction {
	type: CartActionTypes.FETCH_CART_ERROR
	payload: string
}

export type CartAction =
	| FetchCartAction
	| FetchCartSuccessAction
	| FetchCartErrorAction
