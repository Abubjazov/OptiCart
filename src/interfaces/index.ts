export * from './cartListItem.interface'
export * from './product.interface'
export * from './action.interface'
