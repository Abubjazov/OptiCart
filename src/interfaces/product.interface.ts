export enum ProductActionTypes {
	FETCH_PRODUCTS = 'FETCH_PRODUCTS',
	FETCH_PRODUCTS_SUCCESS = 'FFETCH_PRODUCTS_SUCCESS',
	FETCH_PRODUCTS_ERROR = 'FETCH_PRODUCTS_ERROR',
}

export interface Product {
	id: number
	name: string
	picture: string
	picture_medium: string
	picture_small: string
	description: string
	price: number
}

export interface ProductState {
	products: Product[]
	currentProductId: null | number
	status: 'waiting' | 'error' | 'loading'
	error: null | string
}

interface FetchProductsAction {
	type: ProductActionTypes.FETCH_PRODUCTS
}

interface FetchProductsSuccessAction {
	type: ProductActionTypes.FETCH_PRODUCTS_SUCCESS
	payload: Product[]
}

interface FetchProductsErrorAction {
	type: ProductActionTypes.FETCH_PRODUCTS_ERROR
	payload: string
}

export type ProductAction =
	| FetchProductsAction
	| FetchProductsSuccessAction
	| FetchProductsErrorAction
