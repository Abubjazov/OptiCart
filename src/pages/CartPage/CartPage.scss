@import '../../variables';

.cart-page {
	background: $gray;

	.container {
		padding: 11vh 0 0.3vh 0;
		min-height: 87vh;
		max-width: calc(100vw - 10px);
	}

	.cart {
		display: grid;
		grid-template-columns: 1fr;
		row-gap: 15px;
		justify-items: center;
	}

	.total {
		height: 12vh;
		padding: 0 3vw;

		display: flex;
		margin-top: 5px;
		background: $white;
		justify-content: flex-end;
	}
}
