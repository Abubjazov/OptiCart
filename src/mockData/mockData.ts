import { CartListItem } from '../interfaces'

export const cartItems: CartListItem[] = [
	{
		id: 2,
		product_id: 15,
		name: 'Rangoto Vi',
		picture: 'https://i.ibb.co/T2XSnw0/z9.webp',
		description: 'Nemo dolore officia est natus.',
		price: 11.04,
		quantity: 3,
	},
	{
		id: 3,
		product_id: 16,
		name: 'Armo Sento 2',
		picture: 'https://i.ibb.co/TPT93hh/z10.jpg',
		description: 'Nobis in voluptatum qui voluptate.',
		price: 17.22,
		quantity: 1,
	},
	{
		id: 4,
		product_id: 9,
		name: 'Vertigo 2',
		picture: 'https://i.ibb.co/vvCZ6gp/nike.webp',
		description: 'Architecto omnis sed nostrum aliquid.',
		price: 26.11,
		quantity: 2,
	},
]
