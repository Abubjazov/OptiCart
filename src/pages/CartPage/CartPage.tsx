import { nanoid } from 'nanoid'
import { CartItem, Checkout } from '../../components'
import { cartListItem } from '../../interfaces'

import './CartPage.scss'

export const CartPage = (): JSX.Element => {
	const products: cartListItem[] = [
		{
			id: 2,
			product_id: 15,
			name: 'Rangoto Vi',
			picture: 'https://i.ibb.co/T2XSnw0/z9.webp',
			description: 'Nemo dolore officia est natus.',
			price: 11.04,
			quantity: 3,
		},
		{
			id: 3,
			product_id: 16,
			name: 'Armo Sento 2',
			picture: 'https://i.ibb.co/TPT93hh/z10.jpg',
			description: 'Nobis in voluptatum qui voluptate.',
			price: 17.22,
			quantity: 1,
		},
		{
			id: 4,
			product_id: 9,
			name: 'Vertigo 2',
			picture: 'https://i.ibb.co/vvCZ6gp/nike.webp',
			description: 'Architecto omnis sed nostrum aliquid.',
			price: 26.11,
			quantity: 2,
		},
	]

	return (
		<main className='cart-page'>
			<div className='container'>
				<div className='cart'>
					{products.map((item: cartListItem) => (
						<CartItem key={nanoid()} {...item} />
					))}
				</div>
			</div>

			<div className='total'>
				<Checkout />
			</div>
		</main>
	)
}
