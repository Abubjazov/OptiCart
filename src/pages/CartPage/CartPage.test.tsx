import { render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'
import thunk from 'redux-thunk'

import { CartActionTypes } from '../../interfaces'

import { mockCartItems } from '../../mockData/mockData'

import { CartPage } from './CartPage'

const middlewares = [thunk]
const mockStore = configureStore(middlewares)

describe('Component: CartPage', () => {
	test('should render CartPage with Spinner', () => {
		const store = mockStore({
			cart: {
				status: 'loading',
				currentItemId: null,
			},
		})

		const { asFragment } = render(
			<Provider store={store}>
				<CartPage />
			</Provider>
		)
		expect(asFragment()).toMatchSnapshot()
	})

	test('should render CartPage with default store', () => {
		const store = mockStore({
			cart: {
				cart: [],
				status: 'waiting',
			},
		})

		const { asFragment } = render(
			<Provider store={store}>
				<CartPage />
			</Provider>
		)
		expect(asFragment()).toMatchSnapshot()
	})

	test('should render CartPage with cart.status = "loading" & cart.currentItemId != null', () => {
		const store = mockStore({
			cart: {
				cart: mockCartItems,
				status: 'loading',
				currentItemId: 1,
			},
		})

		const { asFragment } = render(
			<Provider store={store}>
				<CartPage />
			</Provider>
		)
		expect(asFragment()).toMatchSnapshot()
	})

	test('should render CartPage with Error', () => {
		const store = mockStore({
			cart: {
				status: 'error',
				currentItemId: 'Test error message',
			},
		})

		const { asFragment } = render(
			<Provider store={store}>
				<CartPage />
			</Provider>
		)
		expect(asFragment()).toMatchSnapshot()
	})

	test('should render CartPage with cart items list', () => {
		const store = mockStore({
			cart: {
				cart: mockCartItems,
				status: 'waiting',
				currentItemId: null,
			},
		})

		const { asFragment } = render(
			<Provider store={store}>
				<CartPage />
			</Provider>
		)
		expect(asFragment()).toMatchSnapshot()
	})

	test('should render CartPage arter Checkout', () => {
		const store = mockStore({
			cart: {
				status: 'checkouted',
			},
		})

		const { asFragment } = render(
			<Provider store={store}>
				<CartPage />
			</Provider>
		)
		expect(asFragment()).toMatchSnapshot()
	})
})
