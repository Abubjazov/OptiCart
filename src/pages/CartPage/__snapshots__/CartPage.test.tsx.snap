// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: CartPage should render CartPage arter Checkout 1`] = `
<DocumentFragment>
  <main
    class="cart-page"
  >
    <div
      class="container"
    >
      <div
        class="message"
        role="log"
      >
        <h2
          tabindex="0"
        >
          Thank you! Your payment has been processed!
        </h2>
        <a
          href="/optiCart/"
        >
          Continue 
          <span>
            Shopping
          </span>
        </a>
      </div>
    </div>
  </main>
</DocumentFragment>
`;

exports[`Component: CartPage should render CartPage with Error 1`] = `
<DocumentFragment>
  <main>
    <div
      class="error-message"
      role="alert"
    >
      <p
        tabindex="0"
      />
      <p
        tabindex="0"
      />
      <a
        href="/optiCart/"
      >
        Back to main page
      </a>
    </div>
  </main>
</DocumentFragment>
`;

exports[`Component: CartPage should render CartPage with Spinner 1`] = `
<DocumentFragment>
  <main
    class="cart-page"
  >
    <div
      class="container"
    >
      <div
        class="cart"
      >
        <svg
          class="spinner"
          height="200"
          preserveAspectRatio="xMidYMid"
          viewBox="0 0 100 100"
          width="200"
        >
          <path
            d="M24.3 30C11.4 30 5 43.3 5 50s6.4 20 19.3 20c19.3 0 32.1-40 51.4-40 C88.6 30 95 43.3 95 50s-6.4 20-19.3 20C56.4 70 43.6 30 24.3 30z"
            fill="none"
            stroke="#3b8079"
            stroke-dasharray="42.76482137044271 42.76482137044271"
            stroke-linecap="round"
            stroke-width="8"
          >
            <animate
              attributeName="stroke-dashoffset"
              dur="1s"
              keyTimes="0;1"
              repeatCount="indefinite"
              values="0;256.58892822265625"
            />
          </path>
        </svg>
      </div>
    </div>
  </main>
</DocumentFragment>
`;

exports[`Component: CartPage should render CartPage with cart items list 1`] = `
<DocumentFragment>
  <main
    class="cart-page"
  >
    <p
      role="alert"
      style="display: block; position: fixed; left: 100px; top: 0px; overflow: hidden; height: 0px; opacity: 1;"
    >
      You are on the cart page
    </p>
    <div
      class="container"
    >
      <div
        class="cart"
      >
         
        <article
          aria-label="Sneakers Indigo in your cart, unit price 16.63 $, quantity 2"
          class="cartitem"
          tabindex="0"
        >
          <header>
            <picture>
              <source
                media="(min-width: 901px)"
                srcset="https://i.ibb.co/6N8qGfs/5-250-250.webp"
              />
              <source
                media="(min-width: 651px)"
                srcset="https://i.ibb.co/FBTQYHv/5-400-400.webp"
              />
              <source
                media="(max-width: 650px)"
                srcset="https://i.ibb.co/6wG8hfd/5-360-360.webp"
              />
              <img
                alt="Indigo"
                src="https://i.ibb.co/6N8qGfs/5-250-250.webp"
              />
            </picture>
            <button
              aria-label="Remove Indigo from your cart"
            >
              Remove product
            </button>
          </header>
          <section
            class="cartitem-descriptor"
          >
            <h2>
              Indigo
            </h2>
            <p>
              Rem ut excepturi quidem nesciunt.
            </p>
          </section>
          <footer>
            <div
              class="quantity"
            >
              <button
                aria-label="Reduce the quantity of Indigo by one"
                class="quantity-minus"
              >
                -
              </button>
              <h3
                aria-label="Quantity of product Indigo is 2"
                class="quantity-value"
                tabindex="0"
              >
                2
              </h3>
              <button
                aria-label="Increase the quantity of Indigo by one"
                class="quantity-plus"
              >
                +
              </button>
            </div>
            <h3
              aria-label="Sneakers Indigo unit price is 16.63 $"
              class="price"
              tabindex="0"
            >
              16.63 $
            </h3>
            <div
              class="full-price"
            >
              <h3
                aria-label="Total price of the ordered sneakers Indigo is 33.26 $"
                tabindex="0"
              >
                33.26 $
              </h3>
            </div>
          </footer>
        </article>
        <article
          aria-label="Sneakers Paco in your cart, unit price 54.5 $, quantity 1"
          class="cartitem"
          tabindex="0"
        >
          <header>
            <picture>
              <source
                media="(min-width: 901px)"
                srcset="https://i.ibb.co/DrymCMM/4-250-250.webp"
              />
              <source
                media="(min-width: 651px)"
                srcset="https://i.ibb.co/84RrCYH/4-400-400.webp"
              />
              <source
                media="(max-width: 650px)"
                srcset="https://i.ibb.co/LSd3drp/4-360-360.webp"
              />
              <img
                alt="Paco"
                src="https://i.ibb.co/DrymCMM/4-250-250.webp"
              />
            </picture>
            <button
              aria-label="Remove Paco from your cart"
            >
              Remove product
            </button>
          </header>
          <section
            class="cartitem-descriptor"
          >
            <h2>
              Paco
            </h2>
            <p>
              Ab odit numquam consectetur aut.
            </p>
          </section>
          <footer>
            <div
              class="quantity"
            >
              <button
                aria-label="Reduce the quantity of Paco by one"
                class="quantity-minus"
              >
                -
              </button>
              <h3
                aria-label="Quantity of product Paco is 1"
                class="quantity-value"
                tabindex="0"
              >
                1
              </h3>
              <button
                aria-label="Increase the quantity of Paco by one"
                class="quantity-plus"
              >
                +
              </button>
            </div>
            <h3
              aria-label="Sneakers Paco unit price is 54.5 $"
              class="price"
              tabindex="0"
            >
              54.5 $
            </h3>
            <div
              class="full-price"
            >
              <h3
                aria-label="Total price of the ordered sneakers Paco is 54.50 $"
                tabindex="0"
              >
                54.50 $
              </h3>
            </div>
          </footer>
        </article>
      </div>
      <div
        class="total"
      >
        <section
          class="checkout"
        >
          <h2
            aria-label="Final price of the entire order is 87.76"
            tabindex="0"
          >
            Total: 87.76 $
          </h2>
          <button
            aria-label="Checkout order"
          >
            Checkout
          </button>
        </section>
      </div>
    </div>
  </main>
</DocumentFragment>
`;

exports[`Component: CartPage should render CartPage with cart.status = "loading" & cart.currentItemId != null 1`] = `
<DocumentFragment>
  <main
    class="cart-page"
  >
    <p
      role="alert"
      style="display: block; position: fixed; left: 100px; top: 0px; overflow: hidden; height: 0px; opacity: 1;"
    >
      You are on the cart page
    </p>
    <div
      class="container"
    >
      <div
        class="cart"
      >
         
        <article
          aria-label="Sneakers Indigo in your cart, unit price 16.63 $, quantity 2"
          class="cartitem"
          tabindex="0"
        >
          <header>
            <picture>
              <source
                media="(min-width: 901px)"
                srcset="https://i.ibb.co/6N8qGfs/5-250-250.webp"
              />
              <source
                media="(min-width: 651px)"
                srcset="https://i.ibb.co/FBTQYHv/5-400-400.webp"
              />
              <source
                media="(max-width: 650px)"
                srcset="https://i.ibb.co/6wG8hfd/5-360-360.webp"
              />
              <img
                alt="Indigo"
                src="https://i.ibb.co/6N8qGfs/5-250-250.webp"
              />
            </picture>
            <button
              aria-label="Remove Indigo from your cart"
            >
              Remove product
            </button>
          </header>
          <section
            class="cartitem-descriptor"
          >
            <h2>
              Indigo
            </h2>
            <p>
              Rem ut excepturi quidem nesciunt.
            </p>
          </section>
          <footer>
            <div
              class="quantity"
            >
              <button
                aria-label="Reduce the quantity of Indigo by one"
                class="quantity-minus"
              >
                -
              </button>
              <h3
                aria-label="Quantity of product Indigo is 2"
                class="quantity-value"
                tabindex="0"
              >
                2
              </h3>
              <button
                aria-label="Increase the quantity of Indigo by one"
                class="quantity-plus"
              >
                +
              </button>
            </div>
            <h3
              aria-label="Sneakers Indigo unit price is 16.63 $"
              class="price"
              tabindex="0"
            >
              16.63 $
            </h3>
            <div
              class="full-price"
            >
              <h3
                aria-label="Total price of the ordered sneakers Indigo is 33.26 $"
                tabindex="0"
              >
                33.26 $
              </h3>
            </div>
          </footer>
        </article>
        <article
          aria-label="Sneakers Paco in your cart, unit price 54.5 $, quantity 1"
          class="cartitem"
          tabindex="0"
        >
          <header>
            <picture>
              <source
                media="(min-width: 901px)"
                srcset="https://i.ibb.co/DrymCMM/4-250-250.webp"
              />
              <source
                media="(min-width: 651px)"
                srcset="https://i.ibb.co/84RrCYH/4-400-400.webp"
              />
              <source
                media="(max-width: 650px)"
                srcset="https://i.ibb.co/LSd3drp/4-360-360.webp"
              />
              <img
                alt="Paco"
                src="https://i.ibb.co/DrymCMM/4-250-250.webp"
              />
            </picture>
            <button
              aria-label="Remove Paco from your cart"
            >
              <svg
                class="small-spinner"
                height="20"
                preserveAspectRatio="xMidYMid"
                viewBox="0 0 100 100"
                width="20"
              >
                <path
                  d="M24.3 30C11.4 30 5 43.3 5 50s6.4 20 19.3 20c19.3 0 32.1-40 51.4-40 C88.6 30 95 43.3 95 50s-6.4 20-19.3 20C56.4 70 43.6 30 24.3 30z"
                  fill="none"
                  stroke="#fff"
                  stroke-dasharray="42.76482137044271 42.76482137044271"
                  stroke-linecap="round"
                  stroke-width="8"
                >
                  <animate
                    attributeName="stroke-dashoffset"
                    dur="1s"
                    keyTimes="0;1"
                    repeatCount="indefinite"
                    values="0;256.58892822265625"
                  />
                </path>
              </svg>
            </button>
          </header>
          <section
            class="cartitem-descriptor"
          >
            <h2>
              Paco
            </h2>
            <p>
              Ab odit numquam consectetur aut.
            </p>
          </section>
          <footer>
            <div
              class="quantity"
            >
              <button
                aria-label="Reduce the quantity of Paco by one"
                class="quantity-minus"
              >
                <svg
                  class="small-spinner"
                  height="20"
                  preserveAspectRatio="xMidYMid"
                  viewBox="0 0 100 100"
                  width="20"
                >
                  <path
                    d="M24.3 30C11.4 30 5 43.3 5 50s6.4 20 19.3 20c19.3 0 32.1-40 51.4-40 C88.6 30 95 43.3 95 50s-6.4 20-19.3 20C56.4 70 43.6 30 24.3 30z"
                    fill="none"
                    stroke="#fff"
                    stroke-dasharray="42.76482137044271 42.76482137044271"
                    stroke-linecap="round"
                    stroke-width="8"
                  >
                    <animate
                      attributeName="stroke-dashoffset"
                      dur="1s"
                      keyTimes="0;1"
                      repeatCount="indefinite"
                      values="0;256.58892822265625"
                    />
                  </path>
                </svg>
              </button>
              <h3
                aria-label="Quantity of product Paco is 1"
                class="quantity-value"
                tabindex="0"
              >
                1
              </h3>
              <button
                aria-label="Increase the quantity of Paco by one"
                class="quantity-plus"
              >
                <svg
                  class="small-spinner"
                  height="20"
                  preserveAspectRatio="xMidYMid"
                  viewBox="0 0 100 100"
                  width="20"
                >
                  <path
                    d="M24.3 30C11.4 30 5 43.3 5 50s6.4 20 19.3 20c19.3 0 32.1-40 51.4-40 C88.6 30 95 43.3 95 50s-6.4 20-19.3 20C56.4 70 43.6 30 24.3 30z"
                    fill="none"
                    stroke="#fff"
                    stroke-dasharray="42.76482137044271 42.76482137044271"
                    stroke-linecap="round"
                    stroke-width="8"
                  >
                    <animate
                      attributeName="stroke-dashoffset"
                      dur="1s"
                      keyTimes="0;1"
                      repeatCount="indefinite"
                      values="0;256.58892822265625"
                    />
                  </path>
                </svg>
              </button>
            </div>
            <h3
              aria-label="Sneakers Paco unit price is 54.5 $"
              class="price"
              tabindex="0"
            >
              54.5 $
            </h3>
            <div
              class="full-price"
            >
              <svg
                class="small-spinner"
                height="45"
                preserveAspectRatio="xMidYMid"
                viewBox="0 0 100 100"
                width="45"
              >
                <path
                  d="M24.3 30C11.4 30 5 43.3 5 50s6.4 20 19.3 20c19.3 0 32.1-40 51.4-40 C88.6 30 95 43.3 95 50s-6.4 20-19.3 20C56.4 70 43.6 30 24.3 30z"
                  fill="none"
                  stroke="#fff"
                  stroke-dasharray="42.76482137044271 42.76482137044271"
                  stroke-linecap="round"
                  stroke-width="3"
                >
                  <animate
                    attributeName="stroke-dashoffset"
                    dur="1s"
                    keyTimes="0;1"
                    repeatCount="indefinite"
                    values="0;256.58892822265625"
                  />
                </path>
              </svg>
            </div>
          </footer>
        </article>
      </div>
      <div
        class="total"
      >
        <section
          class="checkout"
        >
          <svg
            class="small-spinner"
            height="45"
            preserveAspectRatio="xMidYMid"
            viewBox="0 0 100 100"
            width="45"
          >
            <path
              d="M24.3 30C11.4 30 5 43.3 5 50s6.4 20 19.3 20c19.3 0 32.1-40 51.4-40 C88.6 30 95 43.3 95 50s-6.4 20-19.3 20C56.4 70 43.6 30 24.3 30z"
              fill="none"
              stroke="#000"
              stroke-dasharray="42.76482137044271 42.76482137044271"
              stroke-linecap="round"
              stroke-width="3"
            >
              <animate
                attributeName="stroke-dashoffset"
                dur="1s"
                keyTimes="0;1"
                repeatCount="indefinite"
                values="0;256.58892822265625"
              />
            </path>
          </svg>
          <button
            aria-label="Checkout order"
          >
            Checkout
          </button>
        </section>
      </div>
    </div>
  </main>
</DocumentFragment>
`;

exports[`Component: CartPage should render CartPage with default store 1`] = `
<DocumentFragment>
  <main
    class="cart-page"
  >
    <p
      role="alert"
      style="display: block; position: fixed; left: 100px; top: 0px; overflow: hidden; height: 0px; opacity: 1;"
    >
      You are on the cart page
    </p>
    <div
      class="container"
    >
      <div
        class="message"
        role="log"
      >
        <h2
          tabindex="0"
        >
          There are no sneakers in your cart yet!!! )
        </h2>
        <a
          href="/optiCart/"
        >
          Choose 
          <span>
            YOURS!
          </span>
        </a>
      </div>
    </div>
  </main>
</DocumentFragment>
`;
